dockstore-webservice/src/main/resources/migrations.1.17.0.xml

```diff
@@ -38,4 +38,21 @@
             CREATE TRIGGER update_metrics_robot_to_add_privileges BEFORE UPDATE ON enduser FOR EACH ROW WHEN (NEW.metricsrobot AND (NEW.isadmin OR NEW.curator OR NEW.platformpartner IS NOT NULL)) EXECUTE FUNCTION raise_metrics_robot_privileges_exception();
         </sql>
     </changeSet>
+    <changeSet author="svonworl" id="tweakMetricsRobotProperty">
+        <sql dbms="postgresql">
+            DROP TRIGGER update_metrics_robot_to_non_metrics_robot ON enduser;
+            DROP TRIGGER update_metrics_robot_to_add_privileges ON enduser;
+        </sql>
+        <modifyDataType columnName="metricsrobot" tableName="enduser" newDataType="varchar(255)"/>
+        <dropDefaultValue columnName="metricsrobot" tableName="enduser"/>
+        <dropNotNullConstraint columnName="metricsrobot" tableName="enduser"/>
+        <renameColumn oldColumnName="metricsrobot" newColumnName="metricsrobotpartner" tableName="enduser"/>
+        <sql dbms="postgresql">
+            UPDATE enduser SET metricsrobotpartner = CASE WHEN metricsrobotpartner = 'true' THEN 'TOIL' ELSE NULL END;
+        </sql>
+        <sql dbms="postgresql">
+            CREATE TRIGGER update_metrics_robot_to_non_metrics_robot BEFORE UPDATE ON enduser FOR EACH ROW WHEN (OLD.metricsrobotpartner IS NOT NULL AND NEW.metricsrobotpartner IS NULL) EXECUTE FUNCTION raise_metrics_robot_privileges_exception();
+            ALTER TABLE enduser ADD CONSTRAINT metrics_robot_cannot_have_other_privileges CHECK (NOT (metricsrobotpartner IS NOT NULL AND (isadmin OR curator OR platformpartner IS NOT NULL)));
```

Comment:
I'm a little confused. There's a build step called "check that JPA classes are consistent with migrations" 
It checks that the JPA classes match the migrations because it would be weird if they didn't. 
It seems to have run, but this change is only in migrations. 

i.e. its a little weird this check doesn't need to be in jpa annotations OR import.sql

Please add a ticket to follow-up if the script is broken or something

Reply:
I took at look at both `import.sql` and our db/migrations, and it appears there's lots of check constraints, most of which aren't catalogued in `import.sql`.  Indeed, this page suggests that vanilla liquibase doesn't generate diffs for check constraints (see the Liquibase Pro diff types): https://docs.liquibase.com/commands/inspection/diff.html

So, everything appears to be working as expected.